src/sentry/tagstore/legacy/backend.py

```diff
@@ -534,7 +535,8 @@ def get_group_tag_values_for_users(self, event_users, limit=100):
             key='sentry:user',
         ).order_by('-last_seen')[:limit])
 
-    def get_group_ids_for_search_filter(self, project_id, environment_id, tags, limit=1000):
+    def get_group_ids_for_search_filter(
```

Comment:
I will likely follow this up with the wacky self-join implementation that I put together for both backends (see [these examples](https://gist.github.com/tkaemming/57d5ad2e723ce06875536b27e61d6a9a) for a reminder of what that's about), but that seemed better for an independent change since it's fairly self-contained.